### PR TITLE
refactor: calc btc fiat value hook

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5336551178).

Refactoring util hook to accept `Money` type.

### CODE REVIEW

File: src/app/query/common/market-data/market-data.hooks.ts [hunk 1]
```diff
@@ -59,9 +58,9 @@ export function useCalculateBitcoinFiatValue() {
   const btcMarketData = useCryptoCurrencyMarketData('BTC');
 
   return useCallback(
-    (value: string | number) =>
+    (value: Money) =>
       baseCurrencyAmountInQuote(
-        createMoneyFromDecimal(isNumber(value) ? value : Number(value), 'BTC'),
+        createMoneyFromDecimal(value.amount, value.symbol, value.decimals),
```
Comment (fbwoolf) [inline src/app/query/common/market-data/market-data.hooks.ts]: If already `Money`, I don't need this, do I? Not sure I understand the naming of that function?

Comment (fbwoolf) [reply src/app/query/common/market-data/market-data.hooks.ts]: O, I see, this takes the primary unit so `satToBtc` isn't needed? What does `Decimal` refer to in the naming bc both `createMoney` and `createMoneyFromDecimal` use decimals?

File: src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item-with-possible-inscription.tsx [hunk 2]
```diff
@@ -35,15 +35,15 @@ export function PsbtUnsignedInputItemWithPossibleInscription({
...
       path={utxo.inscriptions}
+      value={inputValueAsMoney}
     />
   ) : (
     <PsbtDecodedNodeLayout
       hoverLabel={utxo.address}
       subtitle={truncateMiddle(utxo.address)}
-      subValue={i18nFormatCurrency(calculateBitcoinFiatValue(inputValue))}
-      value={`${isInputCurrentAddress ? '-' : '+'}${inputValue}`}
+      subValue={i18nFormatCurrency(calculateBitcoinFiatValue(createMoney(utxo.value, 'BTC')))}
```
Comment (kyranjamie) [inline src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item-with-possible-inscription.tsx]:
```suggestion
      subValue={i18nFormatCurrency(calculateBitcoinFiatValue(inputValueAsMoney))}
```

File: src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-output-list/components/psbt-unsigned-output-item.tsx [hunk 3]
```diff
@@ -27,14 +27,14 @@ export function PsbtUnsignedOutputItem({
 
   const isOutputCurrentAddress =
     addressFromScript === addressNativeSegwit || addressFromScript === addressTaproot;
-  const outputValue = satToBtc(Number(output.amount)).toString();
+  const outputValueAsMoney = createMoney(Number(output.amount), 'BTC');
```
Comment (kyranjamie) [inline src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-output-list/components/psbt-unsigned-output-item.tsx]:
```suggestion
  const outputValueAsMoney = createMoney(output.amount, 'BTC');
```

cast isn't needed right, we support creating as strings

File: src/app/query/common/market-data/market-data.hooks.ts (hunk 1 above)
Comment (kyranjamie) [reply src/app/query/common/market-data/market-data.hooks.ts]: `createMoneyFromDecimal(value.amount, value.symbol, value.decimals),` you can delete this entire line `baseCurrencyAmountInQuote` accepts money, and it already is that type

Comment (kyranjamie) [reply src/app/query/common/market-data/market-data.hooks.ts]: decimal means you can go `createMoneyFromDecimal(0.1, 'BTC')`, whereas the other accepts sats

File: src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-output-list/components/psbt-unsigned-output-item.tsx (hunk 3 above)
Comment (fbwoolf) [reply src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-output-list/components/psbt-unsigned-output-item.tsx]: This is a bigint.

File: src/app/query/common/market-data/market-data.hooks.ts (hunk 1 above)
Comment (fbwoolf) [reply src/app/query/common/market-data/market-data.hooks.ts]: @kyranjamie saw this prob while you were reviewing, can you comment here on the naming, I don't think the naming of those two functions is clear?

File: src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item-with-possible-inscription.tsx (hunk 2 above)
Comment (fbwoolf) [reply src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item-with-possible-inscription.tsx]: Oops.

File: src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item.tsx [hunk 4]
```diff
@@ -23,14 +23,14 @@ export function PsbtUnsignedInputItem({
...
+  const inputValueAsMoney = createMoney(utxo.value, 'BTC');
 
   return (
     <PsbtDecodedNodeLayout
       hoverLabel={utxo.scriptpubkey_address}
       subtitle={truncateMiddle(utxo.scriptpubkey_address)}
-      subValue={i18nFormatCurrency(calculateBitcoinFiatValue(inputValue))}
-      value={`${isInputCurrentAddress ? '-' : '+'}${inputValue}`}
+      subValue={i18nFormatCurrency(calculateBitcoinFiatValue(inputValueAsMoney))}
+      value={`${isInputCurrentAddress ? '-' : '+'}${inputValueAsMoney.amount.toString()}`}
```
Comment (kyranjamie) [inline src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item.tsx]: we also have the `formatMoney` fn, don't we want this here?

Comment (fbwoolf) [reply src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item.tsx]: We don't want the symbol here.

Comment (fbwoolf) [reply src/app/features/psbt-signer/components/psbt-decoded-request/psbt-unsigned-input-list/components/psbt-unsigned-input-item.tsx]: I could create a new one that doesn't included the symbol? Or, add a prop to include it or not?

File: src/app/common/money/format-money.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import { Money } from '@shared/models/money.model';
 
+const space = ' ';
 const thinSpace = ' ';
 
-export function formatMoney({ amount, symbol, decimals }: Money) {
-  return `${amount.shiftedBy(-decimals).toString()} ${symbol}`;
+export function formatMoney({ amount, symbol, decimals }: Money, includeSymbol = true) {
+  return `${amount.shiftedBy(-decimals).toString()}${includeSymbol ? space + symbol : ''}`;
```
Comment (kyranjamie) [inline src/app/common/money/format-money.ts]: Sorry to nit pick, but can we make this a new function, using boolean [flags as params isn't good readability](https://github.com/ryanmcdermott/clean-code-javascript#dont-use-flags-as-function-parameters)

Comment (fbwoolf) [reply src/app/common/money/format-money.ts]: Np, wasn't sure if should be separate. I'll refactor it.

**Review by kyranjamie — APPROVED** [review]

**Pull request merged**